packages/core/src/schema/columns.ts

```diff
@@ -8,6 +9,43 @@ import type {
   ScalarColumn,
 } from "./common.js";
 
+type DefaultTo<column extends BuilderScalarColumn> = (
+  val: DefaultValue<column[" scalar"]>,
+) => BuilderScalarColumn<
+  column[" scalar"],
+  column[" optional"],
+  column[" list"]
```

Comment:
Need to pass some information about the type of `column[" default"]` to `BuildScalarColumn`